pytorch_lightning/core/lightning.py

```diff
@@ -1757,3 +1736,55 @@ def get_tqdm_dict(self) -> Dict[str, Union[int, str]]:
         rank_zero_warn("`get_tqdm_dict` was renamed to `get_progress_bar_dict` in v0.7.3"
                        " and this method will be removed in v1.0.0", DeprecationWarning)
         return self.get_progress_bar_dict()
+
+    def _auto_register_hparams(self):
+        """
+        Removes the need to pass in hparams. Instead, we register every argument in init
+        to the module with some caveats:
+        1. we don't overwrite the property if it already exists
+        2. we also store a module_arguments property for model loading and saving
+        """
+        # two frames back is the init of the child module
+        frame = inspect.currentframe()
+        args = frame.f_back.f_back.f_locals
+
+        # we'll save hparams automatically (renamed to module_arguments)
+        module_arguments = {}
+
+        # pull out the child itself to make sure we have no issues
+        child = args['self']
+
+        # auto set the attr which enables self.attr anywhere in the code
+        for name, value in args.items():
+
+            # don't add self
+            if name not in ['self']:
+
+                # only track some things
+                is_trackable = self._is_allowed_hparam_value(value)
+
+                # don't overwrite something already set
+                if not hasattr(child, name) and is_trackable:
+                    setattr(child, name, value)
+
+                if is_trackable:
+                    module_arguments[name] = value
+
+        # set module_arguments in child
+        setattr(child, 'module_arguments', module_arguments)
+
+    def _is_allowed_hparam_value(self, value):
```

Comment:
@yukw777 this should be good no? allows for basically anything except objects (but allows dicts, lists, tuples)

Reply:
Previously any picklable objects could be in `hparams`, so I think we should keep that behavior, which is actually quite useful for things like custom vocabulary dictionaries. It also makes it easy to invert dependencies to write tests.

Reply:
@williamFalcon Other examples: this would not allow to use dataclasses and `OmegaConf` which are easy to use now